prodtests/full_system_test.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+#
+# A workflow performing a full system test:
+# - simulation of digits
+# - creation of raw data
+# - reconstruction of raw data
+#
+# Note that this might require a production server to run.
+#
+# This script needs some binary objects (for the moment):
+# - matbud.root + ITSdictionary.bin + ctf_dictionary.root
+#
+# authors: D. Rohr / S. Wenzel
+
+# include jobutils, which notably brings
+# --> the taskwrapper as a simple control and monitoring tool
+#     (look inside the jobutils.sh file for documentation)
+# --> utilities to query CPU count
+. ${O2_ROOT}/share/scripts/jobutils.sh
+
+
+NEvents=${NEvents:-10} #550 for full TF (the number of PbPb events)
+NEventsQED=${NEventsQED:-1000} #35000 for full TF
+NCPUS=$(getNumberOfPhysicalCPUCores)
+echo "Found ${NCPUS} physical CPU cores"
+NJOBS=${NJOBS:-"${NCPUS}"}
+SHMSIZE=128000000000 # 128000000000  # Size of shared memory for messages
+TPCTRACKERSCRATCHMEMORY=22000000000
+NTIMEFRAMES=${NTIMEFRAMES:-1} # Number of time frames to process
+TFDELAY=100 # Delay in seconds between publishing time frames
+MCLABELS=${MCLABELS:-"--disable-mc"}
+
+# allow skipping
+JOBUTILS_SKIPDONE=ON
+
+ulimit -n 4096 # Make sure we can open sufficiently many files
+mkdir qed
+cd qed
+taskwrapper qedsim.log o2-sim -j $NJOBS -n$NEventsQED -m PIPE ITS MFT FT0 FV0 -g extgen --configKeyValues '"GeneratorExternal.fileName=$O2_ROOT/share/Generators/external/QEDLoader.C;QEDGenParam.yMin=-5;QEDGenParam.yMax=6;QEDGenParam.ptMin=0.001;QEDGenParam.ptMax=1.;Diamond.width[2]=6."'
+cd ..
+
+GLOBALDPLOPT="-b --monitoring-backend no-op://"
+taskwrapper sim.log o2-sim -n $NEvents --skipModules ZDC --configKeyValues "Diamond.width[2]=6." -g pythia8hi -j $NJOBS
+taskwrapper digi.log o2-sim-digitizer-workflow -n $NEvents --simPrefixQED qed/o2sim --qed-x-section-ratio 3735  ${MCLABELS} --firstOrbit 0 --firstBC 0 --skipDet TRD --tpc-lanes 36 --shm-segment-size $SHMSIZE ${GLOBALDPLOPT}
```

Comment:
Since you detect the number of CPU cores, I would limit the number of TPC lanes if larger than the number of CPU cores.

Reply:
Done